src/blockdata/opcodes.rs

```diff
@@ -33,573 +32,681 @@ use consensus::encode::{Decodable, Encodable};
-    OP_RETURN_205 = 0xcd,
-    /// Synonym for OP_RETURN
-    OP_RETURN_206 = 0xce,
-    /// Synonym for OP_RETURN
-    OP_RETURN_207 = 0xcf,
-    /// Synonym for OP_RETURN
-    OP_RETURN_208 = 0xd0,
-    /// Synonym for OP_RETURN
-    OP_RETURN_209 = 0xd1,
-    /// Synonym for OP_RETURN
-    OP_RETURN_210 = 0xd2,
-    /// Synonym for OP_RETURN
-    OP_RETURN_211 = 0xd3,
-    /// Synonym for OP_RETURN
-    OP_RETURN_212 = 0xd4,
-    /// Synonym for OP_RETURN
-    OP_RETURN_213 = 0xd5,
-    /// Synonym for OP_RETURN
-    OP_RETURN_214 = 0xd6,
-    /// Synonym for OP_RETURN
-    OP_RETURN_215 = 0xd7,
-    /// Synonym for OP_RETURN
-    OP_RETURN_216 = 0xd8,
-    /// Synonym for OP_RETURN
-    OP_RETURN_217 = 0xd9,
-    /// Synonym for OP_RETURN
-    OP_RETURN_218 = 0xda,
-    /// Synonym for OP_RETURN
-    OP_RETURN_219 = 0xdb,
-    /// Synonym for OP_RETURN
-    OP_RETURN_220 = 0xdc,
-    /// Synonym for OP_RETURN
-    OP_RETURN_221 = 0xdd,
-    /// Synonym for OP_RETURN
-    OP_RETURN_222 = 0xde,
-    /// Synonym for OP_RETURN
-    OP_RETURN_223 = 0xdf,
-    /// Synonym for OP_RETURN
-    OP_RETURN_224 = 0xe0,
-    /// Synonym for OP_RETURN
-    OP_RETURN_225 = 0xe1,
-    /// Synonym for OP_RETURN
-    OP_RETURN_226 = 0xe2,
-    /// Synonym for OP_RETURN
-    OP_RETURN_227 = 0xe3,
-    /// Synonym for OP_RETURN
-    OP_RETURN_228 = 0xe4,
-    /// Synonym for OP_RETURN
-    OP_RETURN_229 = 0xe5,
-    /// Synonym for OP_RETURN
-    OP_RETURN_230 = 0xe6,
-    /// Synonym for OP_RETURN
-    OP_RETURN_231 = 0xe7,
-    /// Synonym for OP_RETURN
-    OP_RETURN_232 = 0xe8,
-    /// Synonym for OP_RETURN
-    OP_RETURN_233 = 0xe9,
-    /// Synonym for OP_RETURN
-    OP_RETURN_234 = 0xea,
-    /// Synonym for OP_RETURN
-    OP_RETURN_235 = 0xeb,
-    /// Synonym for OP_RETURN
-    OP_RETURN_236 = 0xec,
-    /// Synonym for OP_RETURN
-    OP_RETURN_237 = 0xed,
-    /// Synonym for OP_RETURN
-    OP_RETURN_238 = 0xee,
-    /// Synonym for OP_RETURN
-    OP_RETURN_239 = 0xef,
-    /// Synonym for OP_RETURN
-    OP_RETURN_240 = 0xf0,
-    /// Synonym for OP_RETURN
-    OP_RETURN_241 = 0xf1,
-    /// Synonym for OP_RETURN
-    OP_RETURN_242 = 0xf2,
-    /// Synonym for OP_RETURN
-    OP_RETURN_243 = 0xf3,
-    /// Synonym for OP_RETURN
-    OP_RETURN_244 = 0xf4,
-    /// Synonym for OP_RETURN
-    OP_RETURN_245 = 0xf5,
-    /// Synonym for OP_RETURN
-    OP_RETURN_246 = 0xf6,
-    /// Synonym for OP_RETURN
-    OP_RETURN_247 = 0xf7,
-    /// Synonym for OP_RETURN
-    OP_RETURN_248 = 0xf8,
-    /// Synonym for OP_RETURN
-    OP_RETURN_249 = 0xf9,
-    /// Synonym for OP_RETURN
-    OP_RETURN_250 = 0xfa,
-    /// Synonym for OP_RETURN
-    OP_RETURN_251 = 0xfb,
-    /// Synonym for OP_RETURN
-    OP_RETURN_252 = 0xfc,
-    /// Synonym for OP_RETURN
-    OP_RETURN_253 = 0xfd,
-    /// Synonym for OP_RETURN
-    OP_RETURN_254 = 0xfe,
-    /// Synonym for OP_RETURN
-    OP_RETURN_255 = 0xff,
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub struct All {
```

Comment:
If I may make a suggestion:

Now that this is no longer an enum but a struct, the name `All` doesn't make much sense to me. How about renaming it to `OpCode` or something like that?

Reply:
The name `All` is because I wanted to be able to use `opcodes::All` (in contrast to `opcodes::Class` and `opcodes::Ordinary`). I don't see why the restructuring changes that?

I'm not really opposed to using `OpCode` or something, it just feels a bit nicer to me to leave things as is.

Reply:
I thought because the change is already breaking, one might as well tackle this bit. My idea would be to export the struct `OpCode` next to a module `opcodes`, i.e. something like this:

```rust
use bitcoin::blockdata::Opcode;
use bitcoin::blockdata::opcodes::OP_CHECKSIG;
```
star-imports also looks quite nice then:
```rust
use bitcoin::blockdata::opcodes::*;
```
It would have the added benefit that referring to an `OpCode` inside a parameter/struct would be slightly more descriptive:

```rust
fn foo(opcode: OpCode)
```

Reply:
`opcodes::All` and `OpCode` seem exactly as descriptive to me.

I really hate star-imports, they make it impossible to tell where symbols come from, and I don't have any interest in making them easier to use. But sure, we can rename `All` to `OpCode` and re-export it at the crate level.